hathor/nanocontracts/blueprint.py

```diff
@@ -91,6 +110,25 @@ def __new__(cls, name, bases, attrs, **kwargs):
                 # This was not implemented yet and will be extended later.
                 raise BlueprintSyntaxError(f'fields with default values are currently not supported: `{field_name}`')
 
+        # validation makes sure we already have it
+        original_init_fn = getattr(new_class, NC_INITIALIZE_METHOD)
+        init_containers_fn = _make_initialize_uninitialized_container_fields_fn(container_fields)
+
+        # patch initialize method so it initializes containers fields implicitly
+        def patched_init_fn(self: Blueprint, *args: Any, **kwargs: Any) -> None:
+            original_init_fn(self, *args, **kwargs)
+            init_containers_fn(self)
```

Comment:
Why don't we delegate to developers to initialize like they do with regular types?

Reply:
We do, developers can initialize the container fields with this implementation. However, if they don't initialize it in the `initialize` method, it will be _implicitly_ initialized.

This was made in order to support the _current_ behavior of not having to explicitly initialize container fields _and_ to support the behavior of preventing double initializations. We can't have both of these without the implicit initialization.

To prevent double initialization we have to record whether the field was previously initialized. And to support being able to use the field without explicitly initializing it (current behavior), we have to implicitly initialize it (which I chose to do as a post-step injected into the `initialize` method).

Reply:
As discussed, the implicit initialization will be removed in a separate PR.